src/cargo/core/compiler/job_queue.rs

```diff
@@ -583,6 +604,14 @@ impl<'cfg> DrainState<'cfg> {
                     }
                 }
             }
+            Message::FutureIncompatReport(id, report) => {
+                let unit = self.active[&id].clone();
+                self.per_crate_future_incompat_reports
+                    .push(FutureIncompatReportCrate {
+                        package_id: unit.pkg.package_id(),
```

Comment:
I'm not sure I see why this needs to be cloned, maybe something like this?

```suggestion
                let package_id = self.active[&id].pkg.package_id();
                self.per_crate_future_incompat_reports
                    .push(FutureIncompatReportCrate {
                        package_id,
```